clients/csrf/transport.go

```diff
@@ -28,12 +27,13 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 
 	resp, err := t.Delegate.RoundTrip(reqCopy)
 	if err != nil {
-		var clientErr *baseclient.ClientError
-		if errors.As(err, &clientErr) && isCsrfError(clientErr) {
-			csrfTokenManager.invalidateToken()
-		}
 		return nil, err
 	}
+	if isCsrfError(resp) {
+		csrfTokenManager.invalidateToken()
+		return nil, errors.New("Invalid CSRF token error!")
```

Comment:
Go recommends errors messages start with a lower-case letter and do not end with any punctuation (for composition with other error messages)
```suggestion
		return nil, errors.New("invalid CSRF token")
```